### PR TITLE
Provide basic docstring decorator; update goose docstrings

### PR DESCRIPTION
**Pull request by jobrachem**
A basic docstring decorator that only directly reuses docstrings. A more sophisticated version may follow at a later point.

Can be tested and used as follows:

```python
from liesel.docs import usedocs

class ParentClass:
    '''Parentclass documentation'''

    def parent_method(self):
        '''Method documentation.'''

@usedocs(ParentClass)
class InheritingClass:

    @usedocs(ParentClass.parent_method)
    def parent_method(self):
        pass


print(InheritingClass.__doc__)
print(InheritingClass.parent_method.__doc__)
```

### CODE REVIEW

File: liesel/docs.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
+from functools import wraps
+
+
+def usedocs(of: object):
+    """
+    Replaces the docstring of the decorated object with the docstring of the specified
+    object.
+
+    In classes, methods need to be decorated individually. Attribute docstrings are not
+    covered by this decorator.
```
Comment (hriebl) [inline liesel/docs.py:10]: Would it make sense to add a decorator that fills all empty method docstrings of a class with the method docstrings from another class? Maybe that would be useful for the kernel protocols and implementations?

File: liesel/docs.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+    Usage example::
+
+        class ParentClass:
+            '''Parentclass documentation'''
```
Comment (hriebl) [inline liesel/docs.py:15]:
```suggestion
            '''Parent class documentation.'''
```

File: liesel/docs.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+        @usedocs(ParentClass)
+        class InheritingClass:
+
+            @usedocs(ParentClass.parent_method)
+            def parent_method(self):
+                pass
+    """
+
+    def _use_docs(obj):
```
Comment (hriebl) [inline liesel/docs.py:28]:
```suggestion
    def _usedocs(obj):
```

For consistency?

**Review by hriebl — APPROVED** [review]
Just a few minor comments. If you would like to keep things as they are, I'm also fine and ready to merge this.

**Comment by jobrachem** [comment]
@hriebl The decorator is now, as I call it, greedy by default. That means, it will also try to copy method and property docstrings. This can be turned off by using the argument `greedy=False`.

Also, the decorator will never overwrite existing docstrings.

**Comment by jobrachem** [comment]
@GianmarcoCallegher  I converted back to a draft. I'll message you probably tomorrow when the review can actually start.

**Comment by wiep** [comment]
@GianmarcoCallegher, please review and merge (with a squash).

**Comment by hriebl** [comment]
I made a few cosmetic changes. IMO this is ready to be merged.

**Pull request merged**